enterprise/cypher/compatibility-spec-suite/src/test/resources/blacklists/compatibility-23.txt

```diff
@@ -185,3 +185,40 @@ In-query procedure call should fail if one of the argument expressions uses an a
 Comparing strings and integers using > in an AND'd predicate
 Comparing strings and integers using > in a OR'd predicate
 Comparing across types yields null, except numbers
+
+// rule planner/eagerness bug
+Combine MATCH, WITH and CREATE
+
+// TCK side effects are incorrectly specified -- need update
```

Comment:
Does this mean side effects are hard to test for in 2.3, but otherwise these tests would pass? Or does 2.3 have different side effects, and that is perhaps a problem with 2.3 that will not be fixed? In either case is there value in not testing side effects in 2.3?

Reply:
No, these blacklistings (as you touch on in a later comment) are there because the scenarios themselves are incorrect and requires updating. This is done in this PR: https://github.com/opencypher/openCypher/pull/250

Now, the zero-state here is not very different; if you inspect the code currently in `3.3`, side effects are not validated at all (as I mention in the PR description). After introducing correct validation, many scenarios were found to be incorrectly specified (specifies the wrong expectations) and just very few were found to actually pose a problem (bug) in one of the configurations (I believe only in 2.3).

As I see it, there are two viable strategies available here: merge this now and update the TCK dependency as soon as it comes out, effectively un-blacklisting the vast majority of scenarios that have been blacklisted in this PR; or wait with merging this PR until that has happened and make updating the dependency another commit on this PR.

Merging now has the benefit of this PR not falling behind `3.3` and needing rebasing/maintenance. Merging later has the benefit of not blacklisting scenarios, potentially hiding future bugs until the scenarios can be un-blacklisted again. (The scenario is already blacklisted, so introducing a change that makes it "fail more" won't be seen.)